neural_search/util.py

```diff
@@ -251,23 +251,24 @@ def delete_collection(embedding_id: str):
 def detect_outliers(
     project_id: str, embedding_id: str, limit: int = 100
 ) -> Tuple[int, Union[List[Any], str]]:
-    labeled_tensors = embedding.get_manually_labeled_tensors_by_embedding_id(
+    unlabeled_tensors = embedding.get_not_manually_labeled_tensors_by_embedding_id(
```

Comment:
this can be quite a lot of values. (e.g. > 50k) in extreme cases.  should we maybe add a limit (of e.g. 10k) and order by random in selection? Not sure since this will result in values that aren't deterministic.

Reply:
Calculating the outliers on all values can lead to issues quite fast. For each pair of labeled and unlabeled tensors, the distance is calculated. This leads to many calculations, especially if the number of labeled and unlabeled tensors is similar.
I'll try it with 10k, and see if it works for a 50k project.
In that case, I would rather have nondeterministic results than no results.

Reply:
without limit it breaks for 50k project, with 25k records labeled.
With a 10k limit it works, takes 3.5s to calculate on my local machine

Reply:
then lets go with the limit of 10k for now